File: web/stockapi/src/main/scala/zuk/tu_share/dto/TopInst.scala
package zuk.tu_share.dto

import scala.beans.BeanProperty

/***
 * 龙虎榜机构交易单
 */
class TopInst {

  @BeanProperty var trade_date: String = _  //交易日期
  @BeanProperty var ts_code: String = _     //TS代码
  @BeanProperty var exalter: String = _     //营业部名称
  @BeanProperty var buy: String = _         //买入额（万）
  @BeanProperty var buy_rate: String = _    //买入占总成交比例
  @BeanProperty var sell: String = _        //卖出额（万）
  @BeanProperty var sell_rate: String = _   //卖出占总成交比例
  @BeanProperty var net_buy: String = _     //净成交额（万）
  @BeanProperty var side: String = _        //买卖类型0买入1卖出
  @BeanProperty var side_desc: String = _   //
  @BeanProperty var reason: String = _      //上榜理由

  @BeanProperty var count: Int = _          //数量


}

File: web/tushare/hm/top_inst.py
# 导入tushare
import tushare as ts
import pandas as pd
import ZukTuShare

# 初始化pro接口
pro = ZukTuShare.getPro_5000()

# 拉取数据
def top_inst(trade_date):
    df = pro.top_inst(**{
        "trade_date": trade_date,
        "ts_code": "",
        "limit": "",
        "offset": ""
    }, fields=[
        "trade_date",
        "ts_code",
        "exalter",
        "buy",
        "buy_rate",
        "sell",
        "sell_rate",
        "net_buy",
        "side",
        "reason"
    ])
    # print(df)
    return df


if __name__ == '__main__':
    trade_date = "20251218"
    df = top_inst(trade_date)
    df.to_csv(f"top_inst/{trade_date}_top_inst.csv", encoding="utf-8", index=False)
    print("完成")